File: Database/genConvert/JobSetParam.py
def JobSetParam(json):
    this={}#JobSetParamjson)
    #returnfalse
    if 'iname' in json:
        this['iname'] = json['iname']
    if 'job' in json:
        this['job'] = json['job']
    if 'cjob' in json:
        this['jobchange'] = json['cjob']
    if 'target_unit' in json:
        this['target_unit'] = json['target_unit']
    if 'lrare' in json:
        this['lock_rarity'] = json['lrare']
    if 'lplus' in json:
        this['lock_awakelv'] = json['lplus']
    #if(!string.IsNullOrEmpty(json.ljob1))
    #++length
    #if(!string.IsNullOrEmpty(json.ljob2))
    #++length
    #if(!string.IsNullOrEmpty(json.ljob3))
    #++length
    #if(length>0)
        #if(!string.IsNullOrEmpty(json.ljob1))
            this['']
            this['lock_jobs[index]']
            if 'ljob1' in json:
                this['lock_jobs[index]']['iname'] = json['ljob1']
            this['lock_jobs[index]']
            if 'llv1' in json:
                this['lock_jobs[index]']['lv'] = json['llv1']
            #++index
        #if(!string.IsNullOrEmpty(json.ljob2))
            if 'ljob2' in json:
                this['lock_jobs[index]']['iname'] = json['ljob2']
            if 'llv2' in json:
                this['lock_jobs[index]']['lv'] = json['llv2']
            #++index
        #if(!string.IsNullOrEmpty(json.ljob3))
            if 'ljob3' in json:
                this['lock_jobs[index]']['iname'] = json['ljob3']
            if 'llv3' in json:
                this['lock_jobs[index]']['lv'] = json['llv3']
    #returntrue
return this

File: Database/Assembly_SRPG_JP/JobSetParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.JobSetParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

using GR;

namespace SRPG
{
  public class JobSetParam
  {
    public string iname;
    public string job;
    public int lock_rarity;
    public int lock_awakelv;
    public JobSetParam.JobLock[] lock_jobs;
    public string jobchange;
    public string target_unit;

    public bool Deserialize(JSON_JobSetParam json)
    {
      if (json == null)
        return false;
      this.iname = json.iname;
      this.job = json.job;
      this.jobchange = json.cjob;
      this.target_unit = json.target_unit;
      this.lock_rarity = json.lrare;
      this.lock_awakelv = json.lplus;
      this.lock_jobs = (JobSetParam.JobLock[]) null;
      int length = 0;
      if (!string.IsNullOrEmpty(json.ljob1))
        ++length;
      if (!string.IsNullOrEmpty(json.ljob2))
        ++length;
      if (!string.IsNullOrEmpty(json.ljob3))
        ++length;
      if (length > 0)
      {
        this.lock_jobs = new JobSetParam.JobLock[length];
        int index = 0;
        if (!string.IsNullOrEmpty(json.ljob1))
        {
          this.lock_jobs[index] = new JobSetParam.JobLock();
          this.lock_jobs[index].iname = json.ljob1;
          this.lock_jobs[index].lv = json.llv1;
          ++index;
        }
        if (!string.IsNullOrEmpty(json.ljob2))
        {
          this.lock_jobs[index] = new JobSetParam.JobLock();
          this.lock_jobs[index].iname = json.ljob2;
          this.lock_jobs[index].lv = json.llv2;
          ++index;
        }
        if (!string.IsNullOrEmpty(json.ljob3))
        {
          this.lock_jobs[index] = new JobSetParam.JobLock();
          this.lock_jobs[index].iname = json.ljob3;
          this.lock_jobs[index].lv = json.llv3;
          int num = index + 1;
        }
      }
      return true;
    }

    public bool ContainsJob(string iname)
    {
      if (string.IsNullOrEmpty(iname))
        return false;
      if (iname == this.job)
        return true;
      if (this.jobchange == null)
        return false;
      if (iname == this.jobchange)
        return true;
      return MonoSingleton<GameManager>.GetInstanceDirect().MasterParam.GetJobSetParam(this.jobchange).ContainsJob(iname);
    }

    public class JobLock
    {
      public string iname;
      public int lv;
    }
  }
}
